### ISSUE TITLE
Behavior of stdin, stdout, readHandle and toHandle

### ISSUE THREAD

**Issue by k0001** [Gabriel439/Haskell-Pipes-Library#71]
I think there are some issues with the default behavior of `stdin`, `stdout`, `readHandle` and `toHandle`, from the `Pipes.Prelude` module. I'll mention them here so that we can discuss them.
- `readHandle` and `stdin` read an entire line at once, but the produced line doesn't keep the `'\n'` byte. I don't think this is a default sensible behavior given the names of these functions, and the fact that `readHandle` and `stdio` from `Pipes.ByteString` behave differently. Perhaps it should be better to rename `Pipes.Prelude.readHandle` to `Pipes.Prelude.hGetLine` and `Pipes.Prelude.stdin` to `Pipes.Prelude.getLine`, since they behave exactly as the equally named functions in `System.IO`.
  
  If we perform such renaming, then perhaps we don't need to provide functions that behave similarly to `Pipes.ByteString.stdin` and `Pipes.ByteString.readHandle`, since those would be quite inefficient anyway, and `Pipes.Prelude.getLine` and `Pipes.Prelude.hGetLine` should be enough for simple purposes.
  
  In any case, even if I suggest renaming this functions, I would prefer if we just removed these two functions from `Pipes.Prelude`, since `lift`ting `System.IO.getLine` and `System.IO.hGetLine` as `Effect`s should be more than enough.
- `toHandle` and `stdout` share similar problems to the ones mentioned above with regards to line endings. Perhaps renaming them to `hPutStrLn` and `putStrLn`, or just dropping them all together, is better. However, …
- There is a subtle difference between the expected behavior when writing to _standard output_ and when writing to other `Handle`s. Currently, `toHandle` swallows `IOError{ioe_type=ResourceVanished}` exceptions that might happen when trying to write to a `Handle`, gracefully stopping the `Consumer` by returning `()`. However, I think such behavior is appropriate only when the `Handle` is _standard output_. Consider a command-line program invocation like the following:
  
  `foo | head -3`
  
  The `head` program will consume 3 lines of input from `foo`'s output and die, and a `SIGPIPE` signal is sent to the `foo` program. The default behavior in this scenario would be for `foo` to terminate. However, if one decides to ignore the `SIGPIPE` signal (which I think is the case for GHC, but I'm not sure), then the next time `foo` tries to write to _standard output_ it will fail with an exception that matches the following pattern, meaning that we got an _EPIPE_ error:
  
  ``` haskell
  IOError { ioe_type = ResourceVanished
          , ioe_errno = Just ioe }
    | Errno ioe == ePIPE -> ...
  ```
  
  I think it is perfectly fine to swallow such exception and gracefully end a `Consumer` that writes to _standard output_ in that scenario, because:
  - If we are getting that exception it means that the whole program is ignoring `SIGPIPE`, so we are supposed to handle _EPIPE_ errors ourselves.
  - Since getting an _EPIPE_ error when writing to _standard output_ means that the “downstream” program has stopped consuming, and that we should terminate our program as soon as possible (as witnessed by the default behavior of `SIGPIPE`, which is to terminate the program).
  - If we don't swallow such exception, the user executing such program invocation in the command line will see an extraneous “resource vanished” message printed in the screen.
  
  In any case, I think that the GHC RTS might have already solved this issue for us, since in some example code I wrote just now I couldn't manage to reproduce this scenario. I think this issue might be related: http://ghc.haskell.org/trac/ghc/ticket/2699
  
  However, _EPIPE_ errors might also happen when writing to other kinds of `Handle`s that don't behave like _standard output_. For example, when writing to the `Handle` that underlies a TCP connection, you would receive the same kind of exception if the remote end closes its side of the connection, and it is OK to treat that scenario as extraordinary by letting the exception happen, instead of gracefully ending the consumer.
  
  So, in summary: I think `Pipes.Prelude.toHandle` should propagate all exceptions, and `Pipes.Prelude.stdout` should ignore the _EPIPE_ errors. This would give different behaviors between `Pipes.Prelude.toHandle System.IO.stdout` and `Pipes.Prelude.stdout`, but I think that's OK and can be communicated in the documentation. However, I still think we should remove these functions altogether if possible, or at least rename them to `hPutStrLn` and `putStrLn` as I mentioned before.

Notice all that I've mentioned here regarding _EPIPE_ will also apply to the use of output `Handle`s in `Pipes.ByteString`.
